autoscale-core/src/main/java/com/hpe/caf/autoscale/core/ScalerThread.java

```diff
@@ -266,4 +290,33 @@ private int establishMemLimitReached(final double currentMemoryLoad)
         }
         return 0;
     }
+
+    private boolean shouldBackoff()
+    {
+        if (backoff) {
+            backoffCount++;
+            switch (lastOperation) {
+                case SCALE_DOWN: {
+                    return scaleUpBackoffAmount == -1 ? defaultBackoff() : scaleDownBackoffAmount < backoffCount;
+                }
+                case SCALE_UP: {
+                    return scaleUpBackoffAmount == -1 ? defaultBackoff() : scaleUpBackoffAmount < backoffCount;
+                }
+                default: {
+                    return defaultBackoff();
+                }
+            }
+        }
+        return false;
+    }
+
+    private boolean defaultBackoff()
+    {
+        if (backoffCount > backoffAmount) {
+            backoff = false;
+            backoffCount = 0;
+            return false;
```

Comment:
The logic doesn't seem right to me.  Why does these variable resets need to be done in the default case but not in the new cases you've added?